solarforecastarbiter/metrics/probabilistic.py

```diff
@@ -338,3 +338,50 @@ def sharpness(fx_lower, fx_upper):
     """
     sh = np.mean(fx_upper - fx_lower)
     return sh
+
+
+def continuous_ranked_probability_score(fx, fx_prob, obs):
+    """Continuous Ranked Probability Score (CRPS).
+
+        CRPS = 1/n sum_{i=1}^n int |F_i - O_i| dx
+
+    where F_i is the CDF of the forecast at time i and O_i is the CDF
+    associated with the observed value obs_i:
+
+        O_{i, j} = 1 if obs_i <= fx_{i, j}, else O_{i, j} = 0
+
+    where obs_i is the observation at time i, and fx_{i, j} is the forecast at
+    time i for CDF interval j.
+
+    Parameters
+    ----------
+    fx : (n, d) array_like
+        Forecasts (physical units) of the right-hand-side of a CDF with d
+        intervals (d >= 2), e.g., fx = [10 MW, 20 MW, 30 MW] is interpreted as
+        <= 10 MW, <= 20 MW, <= 30 MW.
+    fx_prob : (n, d) array_like
+        Probability [%] associated with the forecasts.
+    obs : (n,) array_like
+        Observations (physical unit).
+
+    Returns
+    -------
+    crps : float
+        The Continuous Ranked Probability Score [unitless].
+
```

Comment:
follow up issue: add a few example plots of F_i and O_i

Reply:
I'll add an examples section here. But regarding the example plots: where would those go? I could try to make ASCII plots in the docstring (which may look okay), add plot code or add PNGs of sample plots to the Metrics page on the website.

Reply:
I had the same questions about the plots which is why I thought it would be best addressed as a separate issue/pr. I think numpy and pandas put plots in the examples sections of their numpy style doc strings, but I've personally tried it.

Reply:
Commit f05ce92 adds an Examples section to the docstring with a couple code examples (inputs and output).

Reply:
I'll add an issue on this topic. (I know what you're referring to and I think I know the main steps, although I too haven't tried it before.)